### DISCUSSION TITLE
Should we move the source code of Fossify to other free software repositories, such as Gitlab or Codeberg?

### DISCUSSION THREAD

**Discussion by csolisr** [FossifyOrg/General-Discussion#37] [Polls]
Since this is a new beginning, it would be a great moment to either move the development of Fossify to a software repository that is not closed-source such as GitHub here. I am aware that a few things would need to be tweaked, such as the GitHub actions to build the APKs automatically if any, but it would be wise to start considering the move from the first days of the project, so that we don't have to deal with that technical debt whenever GitHub is no longer apt to host this project.

**Comment by inson1** [comment]
Github is used by most people, that is fact

But I would like to have mirroring into Codeberg in future (after this project is set and working I'm ready to set some bounties - lets say 2 months?)

**Reply by inson1** [reply]
https://codeberg.org/Recommendations/Mirror_to_Codeberg

**Reply by erebion** [reply]
>Github is used by most people, that is fact

There was a time when most people used Facebook, however, it still was not a good thing.

**Reply by inson1** [reply]
@erebion well, yes and no, it wasnt the best place for people, but company would be crazy to want some online presence and not be on facebook. 
If we move now, we would loose big part of the community and big part the future community.
With mirrors we can see when there is going to be more people on the other platform and move then.
Yes it would be great for Open source, privacy communities etc., but bad for Fossify/SMT community

I think we should help, but not at cost of this project = mirrors

**Comment by tswistak** [comment]
Why wouldn't GitHub be apt to host this project? I'd be more worried about less popular platforms like Codeberg that could disappear any moment than corporate-backed GitHub.

Also, being on GitHub gives better visibility, more people have accounts here and there are already nicely setup issues and discussions (does Gitlab or Codeberg have discussions?). But, just in case, mirror (with turned off issues) can always be done.

**Reply by inson1** [reply]
@tswistak turn on issues - I think they could be mirrored too - I dont know more - @throwaway-d

**Reply by inson1** [reply]
@tswistak  codeberg

> Despite having mirror(s), the project will still continue to be operated from GitHub for development, so having yet another mirror doesn't solve anything.
> 
> A: You can create issues, contribute, view the repos on there without going to GitHub! Example:

**Reply by tswistak** [reply]
But how will it work with templates? Will it be enforced on Codeberg? Also, when issue will be closed on Codeberg, will it also get closed there?

**Reply by inson1** [reply]
@tswistak I think issue closing is mirrored (But I dont know 100%), idk about templates. 
@throwaway-d you could add these 2 questions to your manual :)

**Reply by throwaway-d** [reply]
> But how will it work with templates?

Issue templates, PR templates work as usual.

**Reply by inson1** [reply]
@throwaway-d what about discussions?

**Reply by throwaway-d** [reply]
> what about discussions? 

Currently, Codeberg doesn't have a separate Discussion tab. There are some requests about that:
1. https://codeberg.org/Codeberg/Community/issues/528
2. https://codeberg.org/Codeberg/Community/issues/1331

**Reply by inson1** [reply]
@throwaway-d And is there Option to mirror the whole Org? - it would be easier to do it one time
Thanks for your help :)

**Reply by throwaway-d** [reply]
> And is there Option to mirror the whole Org? - it would be easier to do it one time

It's a requested feature: https://codeberg.org/Codeberg/Community/issues/920

However, people made unofficial script for GitLab you can find in the same link above.

**Reply by erebion** [reply]
>Why wouldn't GitHub be apt to host this project?

One word? Microsoft

**Reply by erebion** [reply]
>more worried about less popular platforms like Codeberg that could disappear any moment than corporate-backed GitHub.

May I remind you of Google Wave and Microsoft Zune?

A few more large projects that were cancelled:

Google Buzz
Google Glass
Googlecatalogs
Google Answers
Google Spaces
Google Reader
Google Wave
iGoogle
Aardvark
Dodgeball
Google Allo
Google Desktop
Google Lively
Google Notebook
Google Play Edition
Nexus Q
Orkut
Google Currents
Google Hangouts
Google Labs

Hosting something yourself gives you control as to how long it will stay up, instead of others deciding for you. Microsoft will not shut down GitHub anytime soon, but that time will come, everything shuts down at some point.

**Reply by tswistak** [reply]
> everything shuts down at some point

Exactly, and I still think it's more probable that other platforms will be sooner shut down than GitHub. I'm not against mirrors, but I think the main repository should still be here. Also, large platforms like GitHub aren't shut down instantly, so there's always time to move. Remember code.google.com? They started closing it in 2013 and completely shut it in 2015. And it wasn't as popular as GitHub is now.

> May I remind you of Google Wave and Microsoft Zune?

Zune devices were never close with popularity to iPods, so it's not unusual that it has been discontinued. But, their marketplace was changed into Xbox/Groove Music, so somehow it still lives. About Google Wave I can't say much - I remember the hype over closed launch and invitations, but later the next thing I've heard was closing it after like a year. It's not comparable to GitHub. 

Additionally, you've listed numerous Google projects, but honestly, most of them weren't as popular (even in their niche) as GitHub is.

**Reply by Hund** [reply]
@tswistak Codeberg uses Forgejo, a fork of Gitea. Which is all open software.

If Codeberg decides to shut down, you can just migrate your project (including all data like issues etc) to your own Forgejo/Gitea instance, or perhaps somewhere else if you don't feel like hosting it yourself. 

If GitHub shuts down, you're screwed.

That's partly what makes open source good. It's in some ways immortal. While proprietary and commercial services like GitHub is fragile, and could potentially die any day.

**Comment by Secret-chest** [comment]
Ideally you would host your own

**Reply by inson1** [reply]
when federation is on

**Comment by erebion** [comment]
Codeberg or another Forgejo instance. Soon we will all be able to participate using federation without being on one single large instance and have more decentralisation. Then we can all (hopefully) finally stop using GitHub and get away from Microsoft.

**Reply by inson1** [reply]
@erebion May you point me to issue, PR or date or something, when the federation will be on? thanks

**Comment by naveensingh** [comment]
We'll do this later.

**Reply by RustoMCSpit** [reply]
is that practical? why set up shop here in the first place?

**Reply by Cwpute** [reply]
mirroring or moving completely can still be done later, what is needed now is just a repo. the project needs to start soon so we don't lose our energy, waiting would just frustrate everyone.

**Reply by naveensingh** [reply]
@RustoMCSpit I meant "mirror".  We won't leave GitHub unless GitHub shutdowns or something better and more popular comes along.

**Comment by Secret-chest** [comment]
If Gitea/Forgejo or GitLab won't suit you, I am developing the roundabout, my own free software git host with plans to become decentralised. But you shouldn't rely on the official instance roundabout-host.com as it is hosted from a Raspberry Pi in my home.
